rule/exported.go

```diff
@@ -446,23 +422,51 @@ func (w *lintExported) lintInterfaceMethod(typeName string, m *ast.Field) {
 		return
 	}
 	name := m.Names[0].Name
-	if !hasDocComment(m.Doc) {
-		w.onFailure(lint.Failure{
-			Node:       m,
-			Confidence: 1,
-			Category:   lint.FailureCategoryComments,
-			Failure:    fmt.Sprintf("public interface method %s.%s should be commented", typeName, name),
-		})
+	firstCommentLine := firstCommentLine(m.Doc)
+	if firstCommentLine == "" {
+		w.addFailure(m, 1, lint.FailureCategoryComments,
+			fmt.Sprintf("public interface method %s.%s should be commented", typeName, name),
+		)
 		return
 	}
-	s := normalizeText(m.Doc.Text())
+
 	expectedPrefix := m.Names[0].Name + " "
-	if !strings.HasPrefix(s, expectedPrefix) {
-		w.onFailure(lint.Failure{
-			Node:       m.Doc,
-			Confidence: 0.8,
-			Category:   lint.FailureCategoryComments,
-			Failure:    fmt.Sprintf(`comment on exported interface method %s.%s should be of the form "%s..."`, typeName, name, expectedPrefix),
-		})
+	if !strings.HasPrefix(firstCommentLine, expectedPrefix) {
+		w.addFailure(m.Doc, 0.8, lint.FailureCategoryComments,
+			fmt.Sprintf(`comment on exported interface method %s.%s should be of the form "%s..."`, typeName, name, expectedPrefix),
+		)
+	}
+}
+
+// mustCheckMethod returns true if the method must be checked by this rule, false otherwise
+func (w *lintExported) mustCheckMethod(fn *ast.FuncDecl) bool {
+	recv := typeparams.ReceiverType(fn)
+
+	if !ast.IsExported(recv) && w.disabledChecks.PrivateReceivers {
+		return false
 	}
+
+	name := fn.Name.Name
+	if commonMethods[name] {
+		return false
+	}
+
+	switch name {
+	case "Len", "Less", "Swap":
+		sortables := w.file.Pkg.Sortable()
+		if sortables[recv] {
+			return false
+		}
+	}
+
+	return true
+}
+
+func (w *lintExported) addFailure(node ast.Node, confidence float64, category lint.FailureCategory, message string) {
+	w.onFailure(lint.Failure{
+		Node:       node,
+		Confidence: confidence,
+		Category:   category,
+		Failure:    message,
+	})
 }
```

Comment:
WDYT?

```suggestion
func (w *lintExported) addFailure(node ast.Node, confidence float64, category lint.FailureCategory, message string, args ...string) {
	w.onFailure(lint.Failure{
		Node:       node,
		Confidence: confidence,
		Category:   category,
		Failure:    fmt.Sprintf(message, args...),
	})
}
```

Reply:
I'm down for it. I would like to suggest renaming the method to addFailuref 

With the logic of sprintf method should ends with an f

Reply:
Indeed!